presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcess.java

```diff
@@ -90,7 +90,7 @@ public NativeExecutionProcess(
     {
         this.port = getAvailableTcpPort();
```

Comment:
Yeah this is kind of tricky. So:
getAvailableTcpPort() checks available ports in the system dynamically and sets the this.port member. 
getPort() is a dummy getter.
at line 93 I need to call getPort() to make use of polymorphism so that the DetachedNativeExecutionProcess can assign a port by itself.